openmmtools/tests/test_integrators.py

```diff
@@ -247,7 +253,76 @@ def test_external_protocol_work_accumulation():
     n_globals = integrator.getNumGlobalVariables()
     names_of_globals = [integrator.getGlobalVariableName(i) for i in range(n_globals)]
     assert('protocol_work' not in names_of_globals), "Protocol work should not be defined."
-
+    del context, integrator
+
+class TestExternalPerturbationLangevinIntegrator(TestCase):
+    def test_protocol_work_accumulation_harmonic_oscillator(self):
+        """Testing protocol work accumulation for ExternalPerturbationLangevinIntegrator with HarmonicOscillator
+        """
+        testsystem = testsystems.HarmonicOscillator()
+        parameter_name = 'testsystems_HarmonicOscillator_x0'
+        parameter_initial = 0.0 * unit.angstroms
+        parameter_final = 10.0 * unit.angstroms
+        for platform_name in ['Reference', 'CPU']:
+            self.compare_external_protocol_work_accumulation(testsystem, parameter_name, parameter_initial, parameter_final, platform_name=platform_name)
+
+    def test_protocol_work_accumulation_waterbox(self):
+        """Testing protocol work accumulation for ExternalPerturbationLangevinIntegrator with AlchemicalWaterBox
+        """
+        from simtk.openmm import app
+        parameter_name = 'lambda_electrostatics'
+        parameter_initial = 1.0
+        parameter_final = 0.0
+        platform_names = [ openmm.Platform.getPlatform(index).getName() for index in range(openmm.Platform.getNumPlatforms()) ]
+        for nonbonded_method in ['CutoffPeriodic', 'PME']:
+            testsystem = testsystems.AlchemicalWaterBox(nonbondedMethod=getattr(app, nonbonded_method))
+            for platform_name in platform_names:
+                name = '%s %s %s' % (testsystem.name, nonbonded_method, platform_name)                
+                self.compare_external_protocol_work_accumulation(testsystem, parameter_name, parameter_initial, parameter_final, platform_name=platform_name, name=name)
+
+    def compare_external_protocol_work_accumulation(self, testsystem, parameter_name, parameter_initial, parameter_final, platform_name='Reference', name=None):
+        """Compare external work accumulation between Reference and CPU platforms.
+        """
+
+        if name is None:
+            name = testsystem.name
+
+        from openmmtools.constants import kB
+        system, topology = testsystem.system, testsystem.topology
+        temperature = 298.0 * unit.kelvin
+        platform = openmm.Platform.getPlatformByName(platform_name)
+
+        # TODO: Set precision and determinism if platform is ['OpenCL', 'CUDA']
+
+        nsteps = 100
```

Comment:
These tests are very slow because `compare_external_protocol_work_accumulation()` is called several times. How about lowering `nsteps` to something like 20?